src/services/catalog.service.ts

```diff
@@ -2,16 +2,75 @@ import * as angular from 'angular';
 import * as _ from 'lodash';
 
 export class CatalogService {
-  static $inject = ['Constants', '$filter'];
+  static $inject = ['$filter', '$q', 'Constants', 'DataService', 'Logger'];
 
   public $filter: any;
-  private constants: any;
+  public catalogServiceItems: any;
+
+  private $q: any;
   private categories: any;
+  private constants: any;
+  private dataService: any;
+  private logger: any;
 
-  constructor(constants: any, $filter: any) {
-    this.constants = constants;
+  constructor($filter: any, $q: any, constants: any, DataService: any, Logger: any) {
     this.$filter = $filter;
+    this.$q = $q;
+    this.constants = constants;
     this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+    this.dataService = DataService;
+    this.logger = Logger;
+  }
+
+  public getCatalogServiceItems() {
+    let deferred: any = this.$q.defer();
+    let serviceClasses: any, imageStreams: any;
+
+    this.getServiceClasses().then((resources: any) => {
+      serviceClasses = resources;
+      this.getImageStreams().then((resources: any) => {
```

Comment:
Better to make these requests asynchronously and use `$q.all()`. It would greatly simplify this logic. I don't think you need to create your own deferred anywhere. Let's chat about this.